tests/core/data/conftest.py

```diff
@@ -8,8 +8,8 @@
 
 def random_numpy_dense(shape):
     """Generate a random numpy dense matrix with the given shape."""
-    out = np.random.rand(*shape) + 1j * np.random.rand(*shape)
-    return out
+    out = np.random.random(shape) + 1j * np.random.random(shape)
+    return np.asarray(out)
```

Comment:
I found quite surprising that 0-dimensional arrays are not keept as ndarrays once they are added together.